cibuild.cmd

```diff
@@ -86,7 +86,22 @@ if defined TestPerfCorrectness (
 
 if defined TestPerfRun (
     msbuild %MSBuildAdditionalCommandLineArgs% Roslyn.sln /p:Configuration=%BuildConfiguration% /p:DeployExtension=false || goto :BuildFailed
-    .\Binaries\%BuildConfiguration%\Roslyn.Test.Performance.Runner.exe --no-trace-upload || goto :BuildFailed
+
+    if defined GIT_BRANCH (
+        REM Check if we have credentials to upload to benchview
+        if defined BV_UPLOAD_SAS_TOKEN (
+            set "EXTRA_PERF_RUNNER_ARGS=--report-benchview --branch "%GIT_BRANCH%""
+
+            REM Check if we are in a PR or this is a rolling submission
+            if defined ghprbPullTitle (
+                set "EXTRA_PERF_RUNNER_ARGS=!EXTRA_PERF_RUNNER_ARGS! --benchview-submission-name "[%ghprbPullAuthorLogin%] PR %ghprbPullId%: %ghprbPullTitle%" --benchview-submission-type private"
+            ) else (
+                set "EXTRA_PERF_RUNNER_ARGS=!EXTRA_PERF_RUNNER_ARGS! --benchview-submission-type rolling"
+            )
+        )
```

Comment:
All of this logic seems very specific to the performance runner tool.  Can we move the logic into that?  

I'm worried about unnecessarily cluttering up the cibuild.cmd file.  It's been getting more and more complicated lately.  Wanted to try and separate out logic if possible.